src/components/EkoSubtitles/EkoSubtitles.jsx

```diff
@@ -0,0 +1,52 @@
+import React, {useEffect, useState} from 'react';
+import "./EkoSubtitles.scss";
+// https://meta.wikimedia.org/wiki/Template:List_of_language_names_ordered_by_code
+const RTL_LANGUAGES = ['ar', 'arc', 'dv', 'fa', 'ha', 'he', 'khw', 'ks', 'ku', 'ps', 'ur', 'yi'];
+const AD_LANGUAGES = ['en-US-AD'];
+
+export function EkoSubtitles({player, style}) {
+    // TODO: Do I need to useRef() or useState() for the player?
+
+    const [visible, setVisible] = useState(false);
+    const [text, setText] = useState('');
+    const [effectiveLang, setEffectiveLang] = useState('');
+
+    useEffect(() => {
+        if (!player) {
+            return;
+        }
+        const onSubtitlesInit = () => player.invoke('subtitles.mode', 'proxy');
+        const onVisibilityChange = (isVisible) =>  setVisible(isVisible);
+        const onSubStart = (subObj) =>  setText(subObj.text);
+        const onSubEnd = (subObj) =>  setText('');
+        const onLangChange = (effectiveLanguage) =>  setEffectiveLang(effectiveLanguage);
+
+        player.on('plugininitsubtitles', onSubtitlesInit);
+        player.on('subtitles.visibilitychange', onVisibilityChange);
+        player.on('subtitles.substart', onSubStart);
+        player.on('subtitles.subend', onSubEnd);
+        player.on('subtitles.effectivelanguagechange', onLangChange);
+        
+        return () => {
+            player.off('plugininitsubtitles', onSubtitlesInit);
+            player.off('subtitles.visibilitychange', onVisibilityChange);
+            player.off('subtitles.substart', onSubStart);
+            player.off('subtitles.subend', onSubEnd);
+            player.off('subtitles.effectivelanguagechange', onLangChange);
```

Comment:
no sure but you might need some safety for the player obj in the case this hook was originally called without the player ref

Reply:
I can add a check to make sure player is defined here! But, I check if the player is defined and return at the top of this fn... so we would never get to this point if the player was undefined, right?

Reply:
Yeah you're completely right. For some reason I had in mind that the the cleanup function isn't defined INSIDE the hook but obvs it'll never get called if the hook itself returns early.

Forget everything that I said then :)